sycl/test/basic_tests/implicit_device_copyable_types.cpp

```diff
@@ -0,0 +1,161 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```

Comment:
I'd like to have more runtime checks that the values of those types actually match between host/device, i.e. check the content of pair.first and/or pair.second.

Reply:
I suggest that we perhaps have to test cases: one which simply contains a bunch of `static_assert(sycl::is_device_copyable_v<some_type_here>)` to check that we properly specialize the trait and another one, which checks runtime behavior, i.e. that we are actually able to properly copy values of those types to kernels and read them back.

Reply:
If this test is supposed to be compile-only, then I suggest to remove `-o %t.out` and add `-fsyntax-only`, see [docs](https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-headers-and-runtime-tests) for some BKMs. You can also remove any code which submits kernels, because that will be checked in the E2E test you have